src/auto-evo/simulation/SimulationCache.cs

```diff
@@ -379,44 +379,78 @@ public void Clear()
         cachedStorageScores.Clear();
     }
 
-    public (float PilusScore, float OxytoxyScore, float MucilageScore) GetPredationToolsRawScores(
+    public (float PilusScore, float OxytoxyScore, float SlimeJetScore) GetPredationToolsRawScores(
         MicrobeSpecies microbeSpecies)
     {
         if (cachedPredationToolsRawScores.TryGetValue(microbeSpecies, out var cached))
             return cached;
 
         var pilusScore = 0.0f;
         var oxytoxyScore = 0.0f;
-        var mucilageScore = 0.0f;
+        var slimeJetScore = 0.0f;
 
         var organelles = microbeSpecies.Organelles.Organelles;
         var organelleCount = organelles.Count;
+        var slimeJetsCount = 0;
+        var topPosition = 0;
+        var bottomPosition = 0;
+        List<int> slimeJetsVerticalPositions = new List<int>();
 
         for (int i = 0; i < organelleCount; ++i)
         {
             var organelle = organelles[i];
 
+            // TODO: Is there better option for calculating/storing cells boundaries?
+            topPosition = Math.Min(organelle.Position.R, topPosition);
+            bottomPosition = Math.Max(organelle.Position.R, bottomPosition);
+
             if (organelle.Definition.HasPilusComponent)
             {
                 pilusScore += Constants.AUTO_EVO_PILUS_PREDATION_SCORE;
                 continue;
             }
 
+            if (organelle.Definition.HasSlimeJetComponent)
+            {
+                slimeJetsVerticalPositions.Add(organelle.Position.R);
+                ++slimeJetsCount;
+                slimeJetScore += Constants.AUTO_EVO_MUCILAGE_PREDATION_SCORE;
+            }
+
             foreach (var process in organelle.Definition.RunnableProcesses)
             {
                 if (process.Process.Outputs.TryGetValue(oxytoxy, out var oxytoxyAmount))
                 {
                     oxytoxyScore += oxytoxyAmount * Constants.AUTO_EVO_TOXIN_PREDATION_SCORE;
                 }
+            }
+        }
 
-                if (process.Process.Outputs.TryGetValue(mucilage, out var mucilageAmount))
-                {
-                    mucilageScore += mucilageAmount * Constants.AUTO_EVO_MUCILAGE_PREDATION_SCORE;
-                }
+        // Make sure that slime jets are positioned at the back of the cell, because otherwise they will push the cell
+        // backwards (into the predator or away from the prey) or to the side
+        foreach (var slimeJetPosition in slimeJetsVerticalPositions)
+        {
+            var mucilageMultiplier = 1.0f;
+            if (bottomPosition != topPosition)
+            {
+                // Range of [0,1] (front to back)
+                mucilageMultiplier = (float)(slimeJetPosition - topPosition) / (bottomPosition - topPosition);
```

Comment:
Isn't this slightly wrong? Like any external organelles the slime jets jump to the closest membrane edge. So the efficiency here should just be calculated as the angle between the Y-axis and a line from the slime jet hex position to the middle hex (which is always 0,0). For examples you can take a look at the movement component movement force calculation.

Reply:
Now the angle between the slimejet and y-axis is calculated instead of this